index.ts

```diff
@@ -418,19 +446,26 @@ export class CapabilitiesController extends BaseController<any, any> implements
     }
   }
 
-  getDomainSettings (domain: string): RpcCapDomainEntry {
-    const domains = this.getDomains();
-
-    // Setup if not yet existent:
-    if (!domains[domain]) {
-      const newDomain = { permissions: [] };
-      domains[domain] = newDomain;
-      return newDomain;
-    }
-
-    return domains[domain];
+  /**
+   * Gets the domain settings for the given IOriginString, or undefined if
+   * none exist.
+   *
+   * @param {IOriginString} domain - The origin string of the domain.
+   * @returns {RpcCapDomainEntry | undefined} - The settings for the domain,
+   * or undefined if none exist.
+   */
+  getDomainSettings (domain: IOriginString): RpcCapDomainEntry | undefined {
+    return this.getDomains()[domain];
```

Comment:
I simplified this function; `getDomainSettings` would _set_ a blank domain setting if none existed for the given domain. That is IMO dangerous, and domain settings now have to be created explicitly using `setDomain`.